### PR TITLE
[flutter_local_notifications] adding ios/macos time sensitive interruption level

### LINKED ISSUE #1495: Support for UNNotificationContent.interruptionLevel on iOS
Hello,

I was trying to see if I could send "time sensitive" and "critical" notifications on iOS using this plugin, but I did not find a way.

I understand that this should be controlled by `UNNotificationContent.interruptionLevel` as described here: https://developer.apple.com/documentation/usernotifications/unnotificationinterruptionlevel

And here:
https://developer.apple.com/design/human-interface-guidelines/ios/system-capabilities/notifications#helping-people-manage-notifications

Can you please confirm that this feature is indeed not implemented? If so, could I help with implementing it?

Great job on the plugin by the way, thank you! 

Cheers,
Marton

### PR DESCRIPTION
**Pull request by maprohu**
Fixes #1495 

Contains changes both for iOS and macOS, based on branch `10.0.0`

Please note my comment in the previous PR, I could not test the notifications on macOS yet. I did not really look into why no notifications show up on my Mac when running the `example` app. Do you happen to have a suggestion where to look?

### CODE REVIEW

**Comment by maprohu** [comment]
I finally managed to test the notifications when running the `example` app as a macOS desktop application. As you suggested, I needed to enable the notifications in System Preferences. 

Note that in order for the time sensitive notifications to work, I had to add the "Time Sensitive Notifications" capability to the macOS project. And that required some change in the signing preferences of the xcode project, and I even ended up registering a new bundle ID for the example app in my Apple developer account. I did not commit these changes, as they are tied to my Apple account. Without making these changes in the xcode project the "time sensitive" setting on notifications are ignored and they behave like normal notifications.

File: flutter_local_notifications/lib/src/platform_specifics/darwin/notification_details.dart [hunk 1]
```diff
@@ -84,4 +86,10 @@ class DarwinNotificationDetails {
   /// This must refer to a [DarwinNotificationCategory] identifier configured
   /// via [InitializationSettings].
   final String? categoryIdentifier;
+
+  /// The interruption level that indicates the priority and
+  /// delivery timing of a notification.
+  ///
+  /// This property is only applicable to iOS 15.0 or newer.
```
Comment (MaikuB) [inline flutter_local_notifications/lib/src/platform_specifics/darwin/notification_details.dart:93]: As this class is for both iOS and macOS, could you add the macOS version too?

**Review by MaikuB — CHANGES_REQUESTED** [review]
Thanks for taking another stab at this. Noticed a couple of minor issues. Rest looks good to me :)

File: flutter_local_notifications/lib/src/platform_specifics/darwin/initialization_settings.dart [hunk 2]
```diff
@@ -9,6 +9,7 @@ class DarwinInitializationSettings {
     this.requestAlertPermission = true,
     this.requestSoundPermission = true,
     this.requestBadgePermission = true,
+    this.requestCriticalPermission = true,
```
[the remark this answers is not in the document]
Comment (maprohu) [reply flutter_local_notifications/lib/src/platform_specifics/darwin/initialization_settings.dart:12]: Good point. I will change the default accordingly.

File: flutter_local_notifications/lib/src/platform_specifics/darwin/notification_details.dart (hunk 1 above)
Comment (maprohu) [reply flutter_local_notifications/lib/src/platform_specifics/darwin/notification_details.dart:93]: Sure, adding comment.

**Comment by MaikuB** [comment]
@maprohu in case it got missed, note that there are conflicts that would need to be resolved as there were changes done. You'll need to update your branch with the latest changes from 10.0.0 to try to resolve them

**Comment by maprohu** [comment]
Hello Michael,
I just fixed the issues that you mentioned above.
Note that I see that there seem to be some conflicts in the branch. Let me know if that something that I should try to fix or you can handle it. 
Cheers,
Marton

**Comment by MaikuB** [comment]
As this is your first PR, I would suggest you can try to see how to fix them as this will probably not be the only time that you'll be submitting PRs in the future :)

From what I could see, they should be relative straightforward to resolve. Main change to note is that recently I made the boolean parameters of the `requestPermissions` method non-nullable and with default values of `false`. That should help you determine what the resolution would be. If you get stuck though then give me a shout and I can help resolve them

**Comment by maprohu** [comment]
@MaikuB 
I think I made a mistake and by accident reverted your changes regarding the `requestPermission` "false by default" parameter. Give me a second...

**Comment by maprohu** [comment]
@MaikuB 
Ok, I think I made it. I do need to step up my merge game... :)

Let me know how it looks!

**Review by MaikuB — APPROVED** [review]

**Comment by MaikuB** [comment]
Looks good to me. I noticed a comment around the ordering of the enum values that I may had missed, good call though made me think that another option is to assign specific values. Something that could be updated later though

**Comment by MaikuB** [comment]
Hmm on further thought, I'll update how the enum is defined before releasing this. For reference, the approach will be similar to how the `Importance` class is defined. Thinking behind this is to make it less prone to errors if the ordering changes even though it's not likely and can help make it more explicit what the value is.

**Comment by maprohu** [comment]
Ok, great, thanks! :)
About the enums, I was also in doubt. Unfortunately (or by design?) Dart enums are rather simple. It would be nice to have instance fields defined for each enum item. 
I agree, the way `Importance` is defined seems to be the next best thing.

**Comment by tripany** [comment]
> I think breaks building for iOS < 15.
> 
> Property 'interruptionLevel' not found on object of type 'UNMutableNotificationContent *'

Scratch that, upgrading from xCode 12 to xCode 13 solved it.

**Pull request merged**